src/Xamarin.Android.Build.Tasks/Tasks/ResolveAssemblies.cs

```diff
@@ -102,10 +101,12 @@ void Execute (MetadataResolver resolver)
 							continue;
 						}
 					}
+					LogDebugMessage ($"Adding {resolved_assembly} to topAssemblyReferences");
 					topAssemblyReferences.Add (resolved_assembly);
 					var taskItem = new TaskItem (assembly) {
 						ItemSpec = Path.GetFullPath (resolved_assembly),
 					};
+					assembly.CopyMetadataTo (taskItem);
```

Comment:
If you use the `TaskItem` constructor that takes a `ITaskItem`, I don't think you need this call?

Docs don't say, but I remember testing that it worked:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.build.utilities.taskitem.-ctor?view=netframework-4.7.2#Microsoft_Build_Utilities_TaskItem__ctor_Microsoft_Build_Framework_ITaskItem_

Reply:
I removed this. It should be happening automagically.